### PR TITLE
Adding an bundlestart event

### PR DESCRIPTION
**Pull request by Tom-Julux**
Plugins should be able to run code before a build process starts.

### CODE REVIEW

File: src/Bundler.js [hunk 1]
```diff
@@ -207,7 +207,8 @@ class Bundler extends EventEmitter {
         });
       });
     }
-
+    this.emit('bundlestart', this.entryFiles)
```
Comment (fathyb) [inline src/Bundler.js]: We currently have a `buildEnd` event when `Bundler` is done, it would be more consistent to name this one `buildStart`

Comment (Tom-Julux) [reply src/Bundler.js]: Definitely makes sense

**Comment by DeMoorJasper** [comment]
Just wondering, this won't block any action, so what's the actual use-case for this?

**Comment by Tom-Julux** [comment]
There are no use cases when running parcel via cli. But for embedding parcel in a larger scope it can be helpful to know when a new build started. For example I use a build server and used this event to set up a webhook.
As this feature is non-blocking, I saw no reason to not implement it. However it is not that important...

**Comment by DeMoorJasper** [comment]
@Tom-Julux Alright :)

I'll merge this in as it makes sense as we already print a message that build starts to cli, makes it available in the api as well :)

**Comment by dsych** [comment]
I am just curious, when can this make it into the official release?

**Comment by DeMoorJasper** [comment]
It’ll be part of 1.9

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**